### DISCUSSION TITLE
Can this library be used to receive an RTP stream without using RTSP?

### DISCUSSION THREAD

**Discussion by Erikmmkarlsson** [AxisCommunications/media-stream-library-js#920] [Q&A] [answered]
In HTML5VideoPipeline you connect to a RTSP server. However, how does it work when receiving an RTP stream only, without using the RTSP protocol?

**Comment by steabert** [comment, marked as answer]
It should be possible to only deal with RTP, but probably not without some modification. In such a case you could skip the RTSP parser component completely, and replace it with an RTP parser. After that you can directly feed a stream of RTP packets to e.g. H.264 depay component.

To make an RTP parser, you could e.g. start from the RTSP parser and remove anything unnecessary.

**Reply by Erikmmkarlsson** [reply]
How does this work in conjunction with websocket? Do you need a websocket proxy to receive the pure rtp stream?
